### PR TITLE
DEScrypt-opencl: Fix for cache files with systemwide builds.

### PR DESCRIPTION
**Pull request by magnumripper**
We need to use JOHN_PRIVATE_HOME for files we write to.
Also, create the opencl subdirectory inside private home directory. This
was needed for the existing shared code as well.

### CODE REVIEW

**Comment by magnumripper** [comment]
Tested now, seem to work just fine.

File: src/opencl_DES_bs_b_plug.c [hunk 1]
```diff
@@ -15,7 +15,12 @@
 #include "../run/opencl/opencl_DES_hst_dev_shared.h"
 #include "mask_ext.h"
 
-#define CONFIG_FILE 	"$JOHN/opencl/DES_bs_kernel_b_%s.config"
+#if JOHN_SYSTEMWIDE
+#define CONFIG_FILE	JOHN_PRIVATE_HOME "/opencl/DES_bs_kernel_b_%s.config"
+#else
+#define CONFIG_FILE	"$JOHN/opencl/DES_bs_kernel_b_%s.config"
```
Comment (solardiz) [inline src/opencl_DES_bs_b_plug.c:21]: I guess it's whitespace changes on the `CONFIG_FILE` line here and in other changed files? If so, I hope they're for the better.

**Review by solardiz — APPROVED** [review]
Makes sense to me. Of course, testing is needed. OK to perform additional testing after merge.

**Comment by solardiz** [comment]
I didn't test this with a system-wide build, but I re-ran the tests of #4670 (as I think it's a platform different from what magnum tested) - no change. Good. I also made sure `descrypt-opencl` actually cracks a password - it did.

File: src/opencl_DES_bs_b_plug.c (hunk 1 above)
Comment (magnumripper) [reply src/opencl_DES_bs_b_plug.c:21]: They were inconsistently made of tabs and/or spaces. All new ones are a single tab between macro name and definition.

**Pull request merged**